client.go

```diff
@@ -92,33 +91,28 @@ func NewClient(options Options) *Client {
 	c := &Client{
 		HTTPClient:  httpclient,
 		HTTPClient2: httpclient2,
-		CheckRetry:  DefaultRetryPolicy(),
-		Backoff:     DefaultBackoff(),
+		CheckRetry:  retrypolicy,
+		Backoff:     backoff,
 		options:     options,
 	}
 
 	c.setKillIdleConnections()
 	return c
 }
 
-// NewWithHTTPClient creates a new Client with default settings and provided http.Client
-func NewWithHTTPClient(client *http.Client, options Options) *Client {
-	httpclient2 := DefaultClient()
-	httpclient2.Transport = client.Transport.(*http.Transport).Clone()
-	if err := http2.ConfigureTransport(httpclient2.Transport.(*http.Transport)); err != nil {
-		return nil
-	}
-	c := &Client{
-		HTTPClient:  client,
-		HTTPClient2: httpclient2,
-		CheckRetry:  DefaultRetryPolicy(),
-		Backoff:     DefaultBackoff(),
+// NewClient creates a new Client with default settings.
+func NewClient(options Options) *Client {
+	return createClient(options, DefaultRetryPolicy(), DefaultBackoff())
+}
 
-		options: options,
-	}
+// NewClient creates a new Client with HTTPErrorRetryPolicy settings.
+func NewClientHTTPRetryPolicy(options Options) *Client {
```

Comment:
I don't think introducing specific retry behaviors is a good approach in a generic library (500 is not assured to be a recoverable error). I think adding the retry policy callback as a field of `Options` is preferable If we want to let the user define a custom retry policy on top of the default one. This way, the custom retry can be passed along with other options as an argument. What do you think?

Reply:
The idea was to give an abstraction, where the user chooses to pick a 500 error code retry policy.
My concern is, passing it as part of option, now gives user multiple ways of configuring retry policy, would not it be confusing? 
How about besides having a factory function to create clients, can we expose a function which let user pass it's retry and backoff policy? Something like this:

`CreateClient(options, retryPolicy, backoff)` 

Let me know your thoughts.

Reply:
What is your opinion about making the `CreateClient` variadic and passing these optional arguments with options like `WithRetryPolicy(...)` and `(WithBackoff(...)`, in this way the signature wouldn't change

Reply:
Sounds good to me! Will do it.